python-libraries/nanover-nglview/src/nanover/nglview/nglclient.py

```diff
@@ -0,0 +1,95 @@
+from io import StringIO
+
+import nglview
+import numpy as np
+
+from nanover.app import NanoverImdClient
+from nanover.mdanalysis import frame_data_to_mdanalysis
+from nanover.trajectory import FrameData
+from nglview import NGLWidget
+
+import MDAnalysis as mda
+
+
+class NGLClient(NanoverImdClient):
+    def __init__(self, dynamic_bonds=False, *args, update_callback=None, **kwargs):
+        self._view = None
+        super().__init__(*args, **kwargs)
+        self.subscribe_to_frames()
+        self.update_callback = update_callback
+        self.dynamic_bonds = dynamic_bonds
+
+    @property
+    def view(self):
+        if self._view is None or self.dynamic_bonds:
+            self._view = frame_data_to_nglwidget(self.latest_frame)
+        return self._view
+
+    def _on_frame_received(self, frame_index: int, frame):
+        super()._on_frame_received(frame_index, frame)
+        self.view.set_coordinates(
+            {0: np.array(self.latest_frame.particle_positions) * 10}
+        )
+        if self.update_callback is not None:
+            self.update_callback(self.universe)
```

Comment:
The type checker doesn't like this because `universe` isn't defined on the class--and I don't see anywhere where it gets set either..

Reply:
I noticed this... I can't tell if this was supposed to add functionality that was never actually completed or whether it does something I'm unaware of... Should we keep it?

Reply:
For now, I have removed this if statement and replaced it with a TODO to implement functionality to allow the user to update callbacks used in the widget.